src/__tests__/formMutations.test.js

```diff
@@ -0,0 +1,330 @@
+/* eslint-env jest */
+/* eslint-disable no-underscore-dangle */
+
+import { connect, disconnect } from '../db/connection';
+import { userFactory, formFactory, formFieldFactory } from '../db/factories';
+import { Forms, Users, FormFields } from '../db/models';
+
+beforeAll(() => connect());
+afterAll(() => disconnect());
+
+describe('form creation tests', () => {
+  let _user;
+  /**
+   * Testing with an _user object
+   */
+  beforeEach(async () => {
+    _user = await userFactory({});
+  });
+
+  /**
+   * Deleting the data that was used in test
+   */
+  afterEach(async () => {
+    await Users.remove({});
+    await Forms.remove({});
+  });
+
+  test('form creation test without userId supplied', async () => {
+    expect.assertions(1);
+    try {
+      await Forms.createForm({
+        title: 'Test form',
+        description: 'Test form description',
+      });
+    } catch (e) {
+      expect(e).toEqual('createdUserId must be supplied');
+    }
+  });
+
+  test('form creating tests', async () => {
+    let form = await Forms.createForm({
+      title: 'Test form',
+      description: 'Test form description',
+      createdUserId: _user._id,
+    });
+
+    form = await Forms.findOne({ _id: form._id });
+
+    expect(form.title).toBe('Test form');
+    expect(form.description).toBe('Test form description');
+    expect(typeof form.code).toBe('string');
+    expect(form.code.length).toEqual(6);
+    expect(typeof form.createdDate).toBe('object');
+    expect(form.createdUserId).toBe(_user._id);
+  });
+});
+
+describe('form update tests', () => {
+  let _user;
+  /**
+   * Testing with an _user object
+   */
+  beforeEach(async () => {
+    _user = await userFactory({});
+  });
+
+  /**
+   * Deleting the data that was used in test
+   */
+  afterEach(async () => {
+    await Users.remove({});
+    await Forms.remove({});
+  });
+
+  test('form update tests', async () => {
+    const form = await Forms.createForm({
+      title: 'Test form',
+      description: 'Test form description',
+      createdUserId: _user._id,
+    });
+
+    await Forms.updateForm(form._id, {
+      title: 'Test form 2',
+      description: 'Test form description 2',
+    });
+
+    const formAfterUpdate = await Forms.findOne({ _id: form._id });
+    expect(formAfterUpdate.title).toBe('Test form 2');
+    expect(formAfterUpdate.description).toBe('Test form description 2');
+    expect(form.createdUserId).toBe(formAfterUpdate.createdUserId);
+    expect(form.code).toBe(formAfterUpdate.code);
+    expect(typeof form.createdDate).toBe('object');
+  });
+});
+
+describe('form remove tests', async () => {
+  let _user;
+  /**
+   * Testing with an _user object
+   */
+  beforeEach(async () => {
+    _user = await userFactory({});
+  });
+
+  /**
+   * Deleting the data that was used in test
+   */
+  afterEach(async () => {
+    await Users.remove({});
+    await Forms.remove({});
+  });
+
+  test('form removal test', async () => {
```

Comment:
When someone try to remove form with some fields, It must raise error. Check this situation